pkg/reconciler/autoscaling/kpa/kpa.go

```diff
@@ -236,7 +239,10 @@ func reportMetrics(pa *pav1alpha1.PodAutoscaler, pc podCounts) error {
 //    | -1   | >= min | >0    | active     | active     |
 func computeActiveCondition(ctx context.Context, pa *pav1alpha1.PodAutoscaler, pc podCounts) {
 	minReady := activeThreshold(ctx, pa)
-	if pc.ready >= minReady {
+	if pc.ready >= minReady ||
+		// For the case of upgrading an existing service (created pre-0.17)
+		// TODO(taragu): remove after 0.18
+		(minReady == 1 && pa.Status.IsInactive() && pa.Status.GetCondition(pav1alpha1.PodAutoscalerConditionActive).Reason == noTrafficReason) {
```

Comment:
Wonder if it's worth pulling this out to a variable, something like `alreadyScaledDownSuccesfully` (the fact Im not totally sure what that variable should be called is why it seems like it may be worth creating one :)).

Reply:
Yeah this represents a pre 0.17 scaled down PA. I'm not sure how to name it either. Maybe `scaledDownPreInitialScalePA`?

Reply:
Maybe we can comment on the fact that the case only happens in 0.17 and focus the variable name on what we're actually searching for here. e.g. maybe

~~~~
// In pre-0.17 we could have scaled down normally without ever setting ScaleTargetInitialized. In this case we'll be in the NoTraffic/inactive state.
alreadyScaledDownSuccesfully = (minReady == 1 && pa.Status.IsInactive() && pa.Status.GetCondition(pav1alpha1.PodAutoscalerConditionActive).Reason == noTrafficReason)
~~~~

Actually, now I've written that, I'm wondering if minReady and Status=Inactive actually matter here? Is there a case where `pa.Status.GetCondition(pav1alpha1.PodAutoscalerConditionActive).Reason == noTrafficReason` and we _dont_ want to set ScaleTargetInitialized?

Reply:
I can't think of a case where we don't want to set ScaleTargetInitialized if `pa.Status.GetCondition(pav1alpha1.PodAutoscalerConditionActive).Reason == noTrafficReason`. @vagababov WDYT?

Reply:
actually minReady=1 is wrong, since minReady with minScale might be > 1.

The corollary is: can we end up _here_ without ever reaching initial scale in 0.17+.

Reply:
ha, right - good point @vagababov. Im pretty sure we just want `pa.Status.GetCondition(pav1alpha1.PodAutoscalerConditionActive).Reason == noTrafficReason` here. I'm pretty sure if we have noTraffic then by definition we scaled up successfully once, because otherwise we'd have TimedOut, and then never becomes NoTraffic, but I'd love someone else to confirm that thinking :D.

Reply:
I think `minReady > 0 && reason == noTrafficReason` together should cover the case.